pkgs/by-name/ea/earlyoom/package.nix

```diff
@@ -24,22 +23,22 @@ stdenv.mkDerivation (finalAttrs: {
 
   patches = [ ./0000-fix-dbus-path.patch ];
 
-  nativeBuildInputs = lib.optionals withManpage [
-    installShellFiles
-    pandoc
-  ];
+  nativeBuildInputs = lib.optionals withManpage [ pandoc ];
 
   makeFlags = [
     "VERSION=${finalAttrs.version}"
+    "DESTDIR=${placeholder "out"}"
```

Comment:
Why those extra lines?

Reply:
To install correctly to `$out` instead of `/usr/bin` and `/etc`. See https://github.com/rfjakob/earlyoom/blob/v1.8.2/Makefile#L38

Previously we write the installation script ourselves in `installPhase` without invoking `make install` at all. But that requires separate maintenance in nixpkgs for duplicated work, especially when upstream include new stuff (like systemd units).

Reply:
Let me open a PR obviating this: 

https://github.com/rfjakob/earlyoom/pull/322

https://github.com/rfjakob/earlyoom/commit/c5a1799a5ff4b3fd3132d50a510e8c126933cf4a.patch

Reply:
Now it was merged

https://github.com/rfjakob/earlyoom/commit/01d8a0e7f4a21d243633f1f7031d442dc28bce0c

Reply:
DESTDIR is not recommended on Nixpkgs.

https://www.gnu.org/prep/standards/html_node/DESTDIR.html

According to GNU Coding Standards, DESTDIR is to be treated as a temporary location.
This is useful for other package distributions that release packages as `.tar.Z` files, but in Nixpkgs we just install everything under PREFIX (eventually splitting things by setting ´MANDIR=$outputMan´ etc.).

Reply:
> in Nixpkgs we just install everything under PREFIX

Fixed. Just that since `MANDIR` is concatenated with `PREFIX` (as you said, it's set to `$out`) so we still need an ugly `/../../..` to redirect it to an absolute `$man` path.

Reply:
I forgot that PREFIX should not be propagated. Here is another patch sent upstream:

https://github.com/rfjakob/earlyoom/pull/324

Reply:
> I forgot that PREFIX should not be propagated. Here is another patch sent upstream:
> 
> [rfjakob/earlyoom#324](https://github.com/rfjakob/earlyoom/pull/324)

Writing makefile manually is hard. :cry:

Reply:
Yay! The patch was merged!

Reply:
Updated according to the new patch.